packages/preferences-persistence/src/index.js

```diff
@@ -38,9 +42,27 @@ export function __unstableCreatePersistenceLayer( serverData, userId ) {
 		preloadedData = convertPreferencesPackageData( serverData );
 	} else if ( localData ) {
 		preloadedData = convertPreferencesPackageData( localData );
-	} else {
+	}
+
+	// Handle legacy data migrations.
+	if ( ! preloadedData ) {
+		// Check if there is data in the legacy format from the old persistence system.
+		const legacyData = getLegacyData( userId );
+		preloadedData = convertLegacyData( legacyData );
+	}
+
+	// The insertUsage preference was migrated later than others, the following code
+	// performs a separate migration just for this data.
+	if ( preloadedData && ! preloadedData?.core?.insertUsage ) {
```

Comment:
Will this run again and again if `insertUsage` is falsey because the user never has inserted a block?

Reply:
Hmmmmm. Yes. I think you're right.

I added an inner `if` statement to check for data to migrate before doing anything more expensive.